### PR TITLE
Add docker image and action for training docker image

### PR DESCRIPTION
**Pull request by darach**
Signed-off-by: Darach Ennis <darach@gmail.com>

# Pull request

Add a docker image and action suitable for learning tremor that derives from base image

## Description

Add a docker image and action suitable for learning tremor that derives from base image

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment

### CODE REVIEW

File: .github/workflows/publish-courseware.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Publish Tremor Courseware Docker image to Registry
+        uses: elgohr/Publish-Docker-Github-Action@master
+        with:
+          name: tremorproject/tremor
```
Comment (Licenser) [inline .github/workflows/publish-courseware.yaml:14]: can we change this to a own image name that way we can tag it and make it clear this is not a production image and shouldn't be used there?

File: .github/workflows/publish-courseware.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Publish Docker - learn
+on:
+  push:
+    branches:
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (Licenser) [inline .github/workflows/publish-courseware.yaml:9]: If insteat of using its own workflow we add this step after the release we can guarnatee we get the right version

File: Dockerfile.learn [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    OPENSSL_STATIC=1
+
+# Switch back to dialog for any ad-hoc use of apt-get
+ENV DEBIAN_FRONTEND=dialog
+ENV RUSTFLAGS="-C target-feature=+avx,+avx2,+sse4.2"
+
+RUN curl https://sh.rustup.rs -sSf | \
+    sh -s -- --default-toolchain stable -y
+ENV PATH=/root/.cargo/bin:$PATH
+RUN cargo install --features=ssl websocat
```
Comment (Licenser) [inline Dockerfile.learn:38]: we should clean up /root/.cargo after this step sop we don't ship the entire intermediate artifacts and rust itself.

**Review by Licenser — CHANGES_REQUESTED** [review]

File: Dockerfile.learn (hunk 3 above)
Comment (darach) [reply Dockerfile.learn:38]: The binary is now moved and the `.cargo` folder removed.

**Review by anupdhml — APPROVED** [review]

**Pull request merged**